libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1893,6 +1893,86 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_mesh2d_get_face_polygons(int meshKernelId, int numNodes, GeometryList& facePolygons)
+    {
+        lastExitCode = meshkernel::ExitCode::Success;
+        try
+        {
+            if (!meshKernelState.contains(meshKernelId))
+            {
+                throw meshkernel::MeshKernelError("The selected mesh kernel id does not exist.");
+            }
+            meshKernelState[meshKernelId].m_mesh2d->Administrate();
+            const auto numFaces = meshKernelState[meshKernelId].m_mesh2d->GetNumFaces();
+            meshkernel::UInt count = 0;
+            for (meshkernel::UInt f = 0u; f < numFaces; ++f)
```

Comment:
no need for u in the assignment. better cast numFaces to UInt.

Reply:
I think that the u (in 0u) is needed, otherwise the 0 will be interpreted as a signed integer.  Now, it could be that some/most/all compilers accept a literal 0, rather than 0u, when assigning to an unsigned int.

auto i = 0; // type int
auto j = 0u; // type unsigned int

i and j will be of different types.

Reply:
gcc 14: accepts with no warning

unsigned int i = 0;